eessi_bot_event_handler.py

```diff
@@ -60,9 +69,137 @@ def handle_issue_comment_event(self, event_info, log_file=None):
+                comment_update += "\n- received bot command "
+                comment_update += f"`{bot_command}`"
+                comment_update += f" from `{sender}` (expanded command format: `{ebc.to_string()}`)"
+            # next two lines are commented out to write less to log files
+            # else:
+                # self.log(f"'{line}' is not considered to contain a bot command")
+                # next three lines commented out to make the bot less noisy,
+                # could be enabled again if bot commands accept arg --verbose
+                # or bot instance has a similar setting
+                # comment_update += f"\n- line <code>{line}</code> is not considered to contain a bot command"
+                # comment_update += "\n- bot commands begin with `bot: `, make sure"
+                # comment_update += "\n  there is no whitespace at the beginning of a line"
+
+        if comment_update == '':
+            # no update to be added, just log and return
+            self.log("update to comment is empty")
+            return
+        else:
+            self.log(f"comment update: '{comment_update}'")
+
+        if not any(map(get_bot_command, comment_update.split('\n'))):
+            # the 'not any()' ensures that the update would not be considered a bot command itself
+            # ... together with checking the sender of a comment update this aims
+            # at preventing the bot to enter an endless loop in commenting on its own
+            # comments
+            update_pr_comment(event_info, comment_update)
+        else:
+            self.log(f"update '{comment_update}' is considered to contain bot command ... not updating PR comment")
+
+        # process commands
+        for cmd in commands:
+            try:
+                update = self.handle_bot_command(event_info, cmd)
+                # the update to the PR comment is only done if the command is
+                # 'help'; otherwise there was too much noise and also updates
+                # were a bit confusing when different instances overwrote
+                # comments of the other instances
+                if cmd.command == 'help':
+                    comment_update += f"\n- handling `{cmd.command}` resulted in: "
+                    comment_update += update
+                    update_pr_comment(event_info, comment_update)
+
+                self.log(f"handling '{cmd.command}' resulted in '{update}'")
+
+            except EESSIBotCommandError as bce:
+                self.log(f"ERROR: handling {cmd.command} failed with {bce.args}")
+                # next two lines commented out to make the bot less noisy,
+                # could be enabled again if bot commands accept arg --verbose
+                # or bot instance has a similar setting
+                # comment_update += f"\n- handling `{cmd.command}` failed with {bce.args}"
+                # update_pr_comment(event_info, comment_update)
```

Comment:
I would post a new comment if an error occurred, these should be visible, otherwise people will get confused that bot is not responding?

Reply:
Yes. This is re-implemented now.